src/app/dynamic-form/components/flex-container/flex-container.component.ts

```diff
@@ -13,37 +9,56 @@ import { DynamicFormControl, FormContainerData, FormControlConfig } from '../../
   templateUrl: './flex-container.component.html',
   styleUrls: ['./flex-container.component.scss']
 })
-export class FlexContainerComponent implements OnInit, AfterViewInit, DynamicFormControl {
-
-  control = new FormGroup({});
+export class FlexContainerComponent implements OnInit, DynamicFormControl {
 
   @Input()
   data: FormContainerData;
 
-  @ViewChildren(ElHostDirective)
-  controls: QueryList<ElHostDirective>;
+  control: FormGroup;
+
+  @ViewChild(ElHostDirective)
+  controlEl: ElHostDirective;
+
+  constructor(private componentFactoryResolver: ComponentFactoryResolver) {
+    this.control = new FormGroup({});
+  }
 
-  constructor(
-    private componentFactoryResolver: ComponentFactoryResolver,
-    private cd: ChangeDetectorRef ) { }
   ngOnInit() {
+    this.generateControls();
   }
 
-  ngAfterViewInit() {
-    const controlsList = this.controls.toArray();
-    this.data.controls.forEach((item: FormControlConfig, index) => {
+  /**
+   * Generates dynamic form controls inserts it inside container.
+   */
+  private generateControls(): void {
+    const viewContainerRef = this.controlEl.containerRef;
+    this.data.controls.forEach((item: FormControlConfig) => {
       const componentFactory = this.componentFactoryResolver.resolveComponentFactory(<any>item.control);
-
-      const viewContainerRef = controlsList[index].containerRef;
-      viewContainerRef.clear();
-
       const componentRef = viewContainerRef.createComponent(componentFactory);
+      viewContainerRef.detach();
+      viewContainerRef.insert(componentRef.hostView);
       const component = <DynamicFormControl>componentRef.instance;
       component.data = item;
-      this.control.addControl(item.name, component.control);
+
+      // Adding control asynchronously to prevent error: ExpressionChangedAfterItHasBeenCheckedError.
+      Promise.resolve().then(() => {
+        this.control.addControl(this.getKey(item), component.control);
+      });
```

Comment:
Used Promise to make changes in `this.control` in next micro task, thats prevents an Error: ExpressionChangedAfterItHasBeenCheckedError.